packages/go/analysis/azure/post.go

```diff
@@ -930,3 +930,33 @@ func UserRoleAssignments(ctx context.Context, db graph.Database) (*analysis.Atom
 		return &operation.Stats, operation.Done()
 	}
 }
+
+// CreateAZRoleApproverEdge ... TBD
+func CreateAZRoleApproverEdge(ctx context.Context, db graph.Database) (graph.NodeSet, error) {
+	var nodeSet graph.NodeSet
+	if err := db.ReadTransaction(ctx, func(tx graph.Transaction) error {
+		var err error
+		if nodeSet, err = ops.FetchNodeSet(tx.Nodes().Filterf(func() graph.Criteria {
+			return query.And(
+				query.Kind(query.Node(), azure.Tenant),
+				query.Kind(query.Node(), azure.Role),
+				query.Equals(
+					query.NodeProperty(azure.EndUserAssignmentRequiresApproval.String()),
+					true,
+				),
+				query.IsNotNull(
+					query.NodeProperty(azure.EndUserAssignmentUserApprovers.String()),
+				),
+			)
+		})); err != nil {
+			return err
+		}
+		// TODO: create edge for AZRoleApprover from the nodeSet
```

Comment:
You can view the NTLM post to view how to do this:

https://github.com/SpecterOps/BloodHound/blob/c0511982c478940ae9411bb9d5aa5d2472181970/packages/go/analysis/ad/ntlm.go#L190

To create the channel

And then pipe in the relationship:
https://github.com/SpecterOps/BloodHound/blob/c0511982c478940ae9411bb9d5aa5d2472181970/packages/go/analysis/ad/ntlm.go#L792-L795

Reply:
There's also some examples in the Azure `post.go`
https://github.com/SpecterOps/BloodHound/blob/7705d4a5bb31b4db4ec1989a2cf3185eb4dad975/packages/go/analysis/azure/post.go#L261-L282